### ISSUE TITLE
Why do we need dracut stuff in the recovery system?

### ISSUE THREAD

**Issue by jsmeix** [rear/rear#3156]
We have in conf/GNU/Linux.conf
```
COPY_AS_IS+=( ... /etc/dracut.conf /etc/dracut.conf.d /usr/lib/dracut ... )
```

It was added via
https://github.com/rear/rear/commit/2d5445466d389b331c9ad1a984c66354415f21ea
which mentiones https://github.com/rear/rear/issues/233
but that does not explain anything about dracut
(or I am blind and don't see the obvious)?

In current GitHub master code
```
find usr/sbin/rear usr/share/rear/ -type f | xargs grep 'dracut'
```
finds only rescue/GNU/Linux/220_load_modules_from_initrd.sh
```
if test -s /etc/dracut.conf ; then
    MODULES_LOAD+=(
        $(
            add_drivers=
            source /etc/dracut.conf
            for s in /etc/dracut.conf.d/*.conf ; do
                source $s
            done
            echo $add_drivers
        )
    )
fi
```
where /etc/dracut.conf and /etc/dracut.conf.d
are used on the original system.

But I fail to see where anything of
/etc/dracut.conf /etc/dracut.conf.d /usr/lib/dracut
is needed in the recovery system.

So I am wondering why we need dracut stuff in the recovery system?

**Comment by jsmeix** [comment]
@rear/contributors 
can someone of you explain to me
whe we we need dracut stuff in the recovery system?

**Comment by schlomo** [comment]
@jsmeix I agree with your question, however I find more files that mention `dracut`:

```
❯ grep -lr dracut
./usr/sbin/rear
./usr/share/rear/rescue/GNU/Linux/220_load_modules_from_initrd.sh
./usr/share/rear/lib/network-functions.sh
./usr/share/rear/finalize/Fedora/550_rebuild_initramfs.sh
./usr/share/rear/finalize/SUSE_LINUX/i386/550_rebuild_initramfs.sh
./usr/share/rear/conf/GNU/Linux.conf
```

However, I also don't see anything that seems to use the dracut configuration files of the rescue system.

I'd suppose this is a leftover from something that either was changed or an attempt that turned out to be not required.

Maybe you just try removing it and check on RHEL and SUSE_LINUX?

Besides that, probably nice example for the value of #3203

**Comment by jsmeix** [comment]
@schlomo 
regarding which files contain 'dracut':
I think things were different at the time when I made
my initial comment. Now I get the same as you:
```
# find usr/sbin/rear usr/share/rear/ -type f | xargs grep -l 'dracut'
usr/sbin/rear
usr/share/rear/conf/GNU/Linux.conf
usr/share/rear/finalize/Fedora/550_rebuild_initramfs.sh
usr/share/rear/finalize/SUSE_LINUX/i386/550_rebuild_initramfs.sh
usr/share/rear/rescue/GNU/Linux/220_load_modules_from_initrd.sh
usr/share/rear/lib/network-functions.sh
```

I will try out how it works on SUSE without
/etc/dracut.conf /etc/dracut.conf.d /usr/lib/dracut
in COPY_AS_IS.

I do not understand why this issue
is an example for the value of
https://github.com/rear/rear/pull/3203
Could you explain?

**Comment by schlomo** [comment]
I thought that the following code could be simplified with the help of a function that reads shell variables from shell-style configuration files:
https://github.com/rear/rear/blob/d3f015f90398c83eb48f6e2a6bc1881da31528ee/usr/share/rear/rescue/GNU/Linux/220_load_modules_from_initrd.sh#L8-L31

**Comment by gdha** [comment]
@jsmeix milestone is still 'ReaR v2.8' - please adjust as you think it best fits (v3.0 or v3.1). Thanks.

**Comment by jsmeix** [comment]
I postponed it to the ReaR v3.1 milestone
because things work with dracut stuff in the recovery system
so this could be done for ReaR v3.1 as time permits.
